### ISSUE TITLE
Client 7.16.3 SubAggregates are not being deserialized 

### ISSUE THREAD

**Issue by lpv90** [elastic/elasticsearch-java#136]
Affects version: `7.16.3`
Doing `filter` aggregation(`status`) with `terms` subAggregation(`status-inner`), ES returns proper JSON with nested subAggregation with buckets:
```
{
    "_shards":
    {
        "failed": 0,
        "skipped": 0,
        "successful": 1,
        "total": 1
    },
    "aggregations":
    {
        "filter#status":
        {
            "doc_count": 12,
            "sterms#status-inner":
            {
                "buckets":
                [
                    {
                        "doc_count": 10,
                        "key": "ACTIVE"
                    },
                    {
                        "doc_count": 2,
                        "key": "SUSPENDED"
                    }
                ],
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0
            }
        }
    },
    "hits":
    {
        "hits":
        [],
        "max_score": null,
        "total":
        {
            "relation": "eq",
            "value": 12
        }
    },
    "timed_out": false,
    "took": 15
}
```
But when the JSON mapped to `SearchResponse`, it only contains top-level `filter` integration:
![image](https://user-images.githubusercontent.com/93690925/151100196-114149fa-f80a-43e4-aa2f-5292c68a850d.png)
So subAggregations are lost during json parsing

**Comment by swallez** [comment]
Thanks for the report @lpv90. This has already been fixed in https://github.com/elastic/elasticsearch-java/issues/87 and will be part of the next release.

**Comment by Scylanth** [comment]
Hello,

As this issue is fixed in the version 7.17 but not backported into version 7.16, is it possible to use the ES client version 7.17 with a version server 7.16.3 ?
